### PR TITLE
Support 8 bi-directional dshot channels on MatekH743

### PR DESCRIPTION
**Pull request by andyp1per**
This also adds DMA to RC input and removes it from SERIAL5 (ESC telemetry) and SERIAL6 to try and balance the sharing

### CODE REVIEW

**Comment by Hwurzburg** [comment]
looking over this, I realized that last month's addition of bdshot for this autopilot was a breaking change for anyone using PPM or SBUS or DSM recievers in a H743 system wishing to upgrade....  @tridge how should we handle this?

**Comment by cbf123** [comment]
Would it be possible to use something similar to get 4 bidirectional DShot channels but full buzzer support?

**Comment by andyp1per** [comment]
No sorry, there is only one timer attached to the buzzer pin

File: libraries/AP_HAL_ChibiOS/hwdef/MatekH743-bdshot/hwdef.dat [hunk 1]
```diff
@@ -5,15 +5,31 @@
...
+undef PC7 PC6 PB0 PB1 PA0 PA1 PA15 PD14 PD15 PE5 PE6 PA2 PA3 PD12 PD13 PB9 PB8 PE0 PE1
 
+# UART4 (spare)
+PB9 UART4_TX UART4 NODMA
+PB8 UART4_RX UART4 NODMA
+
+# UART8 (ESC Telemetry), SERIAL5
+PE0 UART8_RX UART8 NODMA
+PE1 UART8_TX UART8 NODMA
+define HAL_SERIAL5_PROTOCOL SerialProtocol_ESCTelemetry
```
[the remark this answers is not in the document]
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/MatekH743-bdshot/hwdef.dat:17]: It's in the ESC connector - can really only be used for ESC telemetry. Or are you telling me that's not the case on the wing? Feels like we should really have a separate wing target

[the remark this answers is not in the document]
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/MatekH743-bdshot/hwdef.dat:17]: Its is the case for the slim and the mini

[the remark this answers is not in the document]
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/MatekH743-bdshot/hwdef.dat:17]: but the slim pins out besides the ESC connector

[the remark this answers is not in the document]
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/MatekH743-bdshot/hwdef.dat:17]: Sure, but the common use case is someone plugs in a 4-in-1 ESC,. Anyway I am not wedded to it - just trying to make things easier for users.

**Comment by andyp1per** [comment]
This works, but it's very important to have https://github.com/ArduPilot/ardupilot/pull/21976 as well

**Review by tridge — APPROVED** [review]

**Comment by rmackay9** [comment]
I could not cleanly merge this to the Copter-4.3 branch so I've left this out of Copter-4.3.1-rc1 for now

**Comment by rmackay9** [comment]
This is included in Copter-4.3.1-rc1

**Pull request merged**